### ISSUE TITLE
Package schematics / ng update in libraries

### ISSUE THREAD

**Issue by dherges** [ng-packagr/ng-packagr#1180]
## Type of Issue

```
[ ] Bug Report
[x] Feature Request
```

## Description

Currently, ng-packagr only compiles libraries (it was build for component libraries first). A nice addition is to add schematics and ng update rules to the packaging.

The expected output is a `package.json` that contains schematics and ng-update rules (example):

```json
"schematics": "./src/collection.json",
  "ng-update": {
    "requirements": {},
    "packageGroup": [
      "@nrwl/nx",
      "@nrwl/schematics",
      "@nrwl/builders"
    ],
    "migrations": "./migrations/migrations.json"
  },
```

https://github.com/nrwl/nx/blob/master/packages/schematics/package.json#L29-L38

Expected effort: high :)

#### How To Reproduce

Currently, not supported by ng-packagr

#### Expected Behaviour

See above.

#### Version Information

```
$ node_modules/.bin/ng-packagr --version
ng-packagr: 4.5.x
```

**Comment by alan-agius4** [comment]
`ng-add` is another one that should be supported.

**Comment by Chan4077** [comment]
I can try to work on this.

But how would developers declare schematics? Would they do it in the `package.json` file? Or in the `ng-packagr` config?

Maybe allow developers to specify a path to the collection of schematics?

`ng-package.json`

```json
{
  "lib": ...,
  "schematics": "./path/to/collection.json"
}
```

Or maybe allow developers to specify the schematics in the property itself?

`ng-package.json`

```json
{
  "lib": ...,
  "schematics": {
    "ng-add": {
      "description": "NgAdd support ...",
      "factory": "./path/to/index#ngAdd"
    }
  }
}
```

**Comment by kschaefe** [comment]
I agree that ng-packagr should be providing `ng-add` and `ng-update` schematics for all library developers out of the box.  Because I don't do schematics often enough, I either have to look up everything again or copy everything from another library project and adapt it.  Either route takes time away from what I really want to be doing, which is working on the library code.

`ng g library` should offer the ability to create the `ng-add` and `ng-update` schematics for you.  Furthermore, if developers choose not to include them at creation time, they should be able to run a separate schematic later to add them.  Ideally, from a code maintenance and reuse standpoint, the generation schematic should call to the schematics-creating schematics. (It's so meta!)

Running the schematics-creating schematics, via `ng g library` or  manually should alter the library in the following ways:
- Add a `schematics` directory that is a sibling of `src` in the `projeects/library-name`.  That directory will contains
    - `collection.json`
    - the `ng-add` directory
    - the `ng-update` directory
- Modify `package.json` (not `ng-package.json`).  The latter is not read by `ng add <package>`.
    - add `"schematics": "./schematics/collection.json"`
    - add scripts for building, testing, and packaging the schematics with the library.  There will be no perfect answer for this section.  My company likes to call `ng build` directly, others use `npm run build` scripts for that.  `ng-packagr` should choose an option that makes sense and document it.

Once these prerequisite schematics are in place, `ng-packagr` will be a in position to add schematics for other tasks that would need to modify `ng-add` and `ng-update`.  For instance, [copying assets](https://github.com/ng-packagr/ng-packagr/blob/master/docs/copy-assets.md) requires a change to `angular.json` of any application using the library.  Such modification is a perfect task for `ng-add` for new installations or `ng-update` for updating the library.

I actually came here to file a feature request about that specific use-case (asset copying) for `ng-add`/`ng-update`.  I am happy to expand that into its own top-level feature request now or in the future, but it seemed logical to add it as an example of what `ng-packagr` could be doing for library developers in this schematics request.
